docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,199 @@
+
+	private readonly HttpClient _httpClient = new HttpClient();
+
+	[HttpGet]
+	public async Task<int> MicrosoftCount()
+	{
+	    // Suspends MicrosoftCount() to allow the caller (the web service) to accept another request,
+	    // rather than blocking on this one.
+	    var html = await _httpClient.DownloadStringAsync("microsoft.com");
+
+	    return Regex.Matches(html, "Microsoft").Count;
+	}
+
+Example (VB)
+------------
+
+These are the VB-equivalent code snippets from above.
+
+Client app snippet (Universal Windows App):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	Private Async Sub SeeTheMicrosofts_Click(sender As Object, e As EventArgs)
+
+	    ' Capture the task handle here so we can await it later.
+	    Dim getMicrosoftDotComHtmlTask As Task(Of String) = _httpClient.GetStringAsync("microsoft.com")
+
+	    ' Any other work on the UI thread can be done here, such as enabling a Progress Bar.
+	    NetworkProgressBar.IsEnabled = true
+	    NetworkProgressBar.Visibility = Visibility.Visible
+
+	    ' The await operator suspends SeeTheMicrosofts_Clicked, returning control to its caller.
+	    ' This is what allows the app to be responsive and not hang on the UI thread.
+	    Dim html As String = Await getMicrosoftDotComHtmlTask
+	    Dim count As Integer = Regex.Matches(html, "Microsoft").Count
+
+	    MicrosoftCountLabel.Text = $"Number of Microsofts on microsoft.com: {count}"
+
+	    NetworkProgressBar.IsEnabled = false
+	    NetworkProgressBar.Visbility = Visibility.Collapsed
+	End Sub
+
+Web Service snippet (ASP.NET MVC):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	<HttpGet>
+	Public Async Function MicrosoftCount() As Task(Of String)
+
+	    ' Suspends MicrosoftCount() to allow the caller (the web service) to accept another request,
+	    ' rather than blocking on this one.
+	    Dim html As String = Await _httpClient.GetStringAsync("microsoft.com");
+
+	    Return Regex.Matches(html, "Microsoft").Count
+	End Function
+
+More on Task and Task<T>
+------------------------
+
+As mentioned before, Tasks are constructs used to represent operations which you will need to wait on to complete.  Although Tasks are general, the important context here is using them to represent an I/O operation.
+
+* ``Task`` represents a single operation which does not return a value.
+* ``Task<T>`` represents a single operation which returns a value of type ``T``.
+
+It's important to reason about Tasks as abstractions of work to be done rather than threads, because in the context of asynchronous I/O they aren't threads at all!  Asynchronous .NET library calls essentially delegate work to the operating system, which is naturally asynchronous when I/O is performed.  Unless you write code using a Task to abstract the notion of a new thread (such as the ``Task.Run`` method), there will be no new thread spawned as a result of your async code.
```

Comment:
> They are not ever threads.

`Task.Run` specifically does create a new thread, and is a recommended pattern to follow when you have CPU-bound work.  In that particular case, the task handle returned is representing a new thread.  This is why I'm stating that it's in the context of I/O.

> Lots of .NET APIs delegate work to the OS. In this case, the delegation is to a low-level file system driver that calls back when there is work to do. My understanding is that the file system has both sync and async modes. Assuming thats true, we should avoid the word "naturally".

I don't know about Windows internals that much, but IIRC in my operating systems class I read that internally, I/O requests are asynchronous and drivers are not allowed to block on I/O requests (unless the request can be processed *immediately* which is rarely the case).  On Linux that convention is followed as well, even if the code making the POSIX call is synchronous.  Especially if Linux is going to be a huge target system to focus on in these docs, it's very much a world where async, signals, and interrupts are the norm.